tweetflask.py

```diff
@@ -0,0 +1,114 @@
+from flask import Flask, jsonify, request
+from flask_basicauth import BasicAuth
+from datetime import datetime
+
+import json
+import yaml
+import ssl
+
+
+app = Flask(__name__)
+
+
+@app.route('/tweets')
+def get():
+    # Opening and reading the db file
+    with open('db.json', 'r') as jfile:
+        jsondata = json.load(jfile)
+    # Parsing the username and mood from url
+    username = request.args.get('username')
+    mood = request.args.get('mood')
+    # return 400 for not providing a username
+    if not username:
+        return jsonify(errors('400', 'No username provided')), 400
+    # looking up the username in the DB
+    for user in jsondata['UserData']:
+        if username.upper().lower() == user['Username'].upper().lower():
+            return jsonify(get_tweet(jsondata, user, mood))
+    # if the username does not exist in DB this line will be excuted
+    # to return an empty array
+    return jsonify(get_tweet(jsondata, None, None))
+
+
+# looks up all the tweets that belong to the username provided
+# and looks for the mood if provided
+def get_tweet(jsondata, user, mood):
+    tweetdata = {
+        'data': [
+        ]
+    }
+    if user is None:
+        return tweetdata
+    for tweets in jsondata['TweetData']:
+        if user['UserID'] == tweets['UserID']:
+            if not mood or mood.lower() == tweets['Mood'].lower():
+                response_data = {}
+                response_data['id'] = tweets['TweetID']
+                response_data['type'] = "tweet"
+                time = datetime.strptime(tweets['TimeAndDate'],
+                                         '%d-%b-%y %I.%M.%S.%f %p %z')
```

Comment:
I sent a request to `https://localhost:5000/tweets?username=Alawammo` but got the following error:

```
ValueError: time data '23-OCT-18 01.18.44.281602 PM -07:00' does not match format '%d-%b-%y %I.%M.%S.%f %p %z'
```

Could you take a look?

Reply:
Feel free to ignore this comment. This issue has been solved in Python 3.7+.